source/ed/connectors/osm_tags_reader.cpp

```diff
@@ -224,4 +226,55 @@ std::bitset<8> parse_way_tags(const std::map<std::string, std::string> & tags){
     return result;
 }
 
-}} //namespace navitia::OSM
+PoiTypeParams::PoiTypeParams(const std::string& json_params) {
+    namespace pt = boost::property_tree;
+
+    pt::ptree poi_root;
+    std::istringstream iss (json_params);
+    pt::read_json(iss, poi_root);
+
+    for (const pt::ptree::value_type& json_poi_type: poi_root.get_child("poi_types")) {
+        std::string poi_type_id = json_poi_type.second.get<std::string>("id");
+        if (poi_types.find(poi_type_id) != poi_types.end()) {
+            throw std::invalid_argument("POI type id " + poi_type_id + " is defined multiple times");
+        }
+        poi_types[poi_type_id] = json_poi_type.second.get<std::string>("name");
+    }
+    if (poi_types.find("amenity:parking") == poi_types.end() ||
+        poi_types.find("amenity:bicycle_rental") == poi_types.end()) {
+        throw std::invalid_argument("The 2 POI types id=amenity:parking and "
+                                    "id=amenity:bicycle_rental must be defined");
+    }
+
+    for (const pt::ptree::value_type& json_rule: poi_root.get_child("rules")) {
+        rules.emplace_back();
+        auto& rule = rules.back();
+        rule.poi_type_id = json_rule.second.get<std::string>("poi_type_id");
+        if (poi_types.find(rule.poi_type_id) == poi_types.end()) {
+            throw std::invalid_argument("Using an undefined POI type id (" + rule.poi_type_id + ") in rules");
```

Comment:
minor: you may want to throw the exception **before** you create a new element in `rules`